src/singularsamplebuffer.h

```diff
@@ -0,0 +1,85 @@
+#ifndef SINGULARSAMPLEBUFFER_H
+#define SINGULARSAMPLEBUFFER_H
+
+#include "samplebuffer.h"
+
+// A singular FIFO/LIFO sample buffer with fixed capacity and range
+// checking.
+//
+// Common use case: Consume all buffered samples before the capacity
+// is exhausted.
+//
+// This class is not thread-safe and not intended to be used from multiple
+// threads!
+class SingularSampleBuffer {
+    Q_DISABLE_COPY(SingularSampleBuffer);
+
+public:
+    SingularSampleBuffer();
+    explicit SingularSampleBuffer(SINT capacity);
+    virtual ~SingularSampleBuffer() {}
+
+    // The initial/total capacity of the buffer.
+    SINT getCapacity() const {
+        return m_primaryBuffer.size();
+    }
+
+    // The capacity at the tail that is immediately available
+    // without trimming the buffer.
+    SINT getTailCapacity() const {
```

Comment:
I am confused with the tail/head nomenclature. 
It is possible to turn it into read/write?